### PR TITLE
Celene/static tags collector

### PR DESCRIPTION
**Pull request by celenechang**
### What does this PR do?
Adds a simple FetchOnly static tags collector that can be used for tags that do not change after start, e.g. those from environment variables.

Adds `DD_TAGS` env var extract to static collector and removes this functionality from the ECS fargate collector.

### Motivation

This allows `DD_TAGS` to be applied in environments where specialized collectors were not present such as in EKS Fargate.

### Describe your test plan

- Ensure `DD_TAGS` are applied when monitored pods and containers are started or cycled
- Ensure `DD_TAGS` are applied in ECS and EKS Fargate
- Ensure `DD_TAGS` are part of host tags and not output of `agent tagger-list` for non-EKS clusters

### CODE REVIEW

File: pkg/tagger/collectors/static_extract.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-2020 Datadog, Inc.
+
+// +build docker kubelet
```
Comment (xornivore) [inline pkg/tagger/collectors/static_extract.go:6]: Do we need to guard with these build flags? Does look pretty generic here? Maybe some other dependency requirement to have these build flags on?

File: pkg/tagger/utils/taglist.go [hunk 2]
```diff
@@ -74,8 +74,8 @@ func (l *TagList) AddAuto(name, value string) {
 	l.AddLow(name, value)
 }
 
-// Compute returns two string arrays in the format "tag:value"
-// first array is low cardinality tags, second is high card ones
+// Compute returns three string arrays in the format ["tag0:value0", "tag1:value1"]:
```
Comment (xornivore) [inline pkg/tagger/utils/taglist.go:77]: Should we convert return value into a struct?

File: pkg/tagger/collectors/static_main.go [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+	expire       *utils.Expire
+	lastExpire   time.Time
+	expireFreq   time.Duration
+	ddTagsEnvVar []string
+}
+
+func (c *StaticCollector) Detect(out chan<- []*TagInfo) (CollectionMode, error) {
+	var err error
+	c.infoOut = out
+	c.expire, err = utils.NewExpire(staticExpireFreq)
```
Comment (xornivore) [inline pkg/tagger/collectors/static_main.go:36]: Are these needed? Mind clarifying how expiration works here?

File: pkg/tagger/collectors/static_main.go [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+	c.ddTagsEnvVar = config.Datadog.GetStringSlice("tags")
+
+	if err != nil {
+		return FetchOnlyCollection, fmt.Errorf("Failed to instantiate the expiration process")
+	}
+	return FetchOnlyCollection, nil
+}
+
+// Fetch fetches static tags
+func (c *StaticCollector) Fetch(entity string) ([]string, []string, []string, error) {
```
Comment (xornivore) [inline pkg/tagger/collectors/static_main.go:49]: Having a struct as returned value might make it clearer here.

**Review by xornivore — COMMENTED** [review]
Looks great! Few small comments.

File: pkg/tagger/collectors/static_main.go (hunk 3 above)
Comment (celenechang) [reply pkg/tagger/collectors/static_main.go:36]: you're right, they aren't needed - i pulled out the ComputeExpires part and forgot about this

File: pkg/tagger/utils/taglist.go (hunk 2 above)
Comment (celenechang) [reply pkg/tagger/utils/taglist.go:77]: perhaps i'll make a follow up PR to update this everywhere, OK with you?

Comment (xornivore) [reply pkg/tagger/utils/taglist.go:77]: yes, sure, np

File: pkg/tagger/collectors/static_extract.go (hunk 1 above)
Comment (celenechang) [reply pkg/tagger/collectors/static_extract.go:6]: i had in the back of my mind any potential downsides to having `DD_TAGS` picked up as host tags as well as the static collector, but even then this isn't the right way to guard against that. going to take ahmed's suggestion and only use this check on fargate, as those are the situations where the hostname is empty and thus these tags aren't added to the metadata

File: pkg/tagger/collectors/static_main.go [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+	return c.ddTagsEnvVar, []string{}, []string{}, nil
+}
+
+func staticFactory() Collector {
+	return &StaticCollector{}
+}
+
+func init() {
+	// Only register collector if it is a Fargate instance
```
Comment (xornivore) [inline pkg/tagger/collectors/static_main.go:48]: Mind adding that this applies to both ECS and EKS Fargate?

**Review by xornivore — APPROVED** [review]
👍 just a small ask to tweak a comment for clarity

**Pull request merged**